torch/fx/experimental/proxy_tensor.py

```diff
@@ -470,6 +465,25 @@ def dispatch_trace(
     return GraphModule(tracer.root, graph, name)
 
 
+@contextlib.contextmanager
+def _pop_proxy_mode_temporarily(dk):
+    # This is a shim around the existng per-dispatch-key-mode logic.
+    # I'll delete the per-dispatch-key-mode logic in a followup PR
+    if dk is not None:
+        # During pre_dispatch, pop off of the PreDispatch mode stack
+        old = _pop_mode(dk)
+        try:
+            yield old
+        finally:
+            _push_mode(old, dk)
```

Comment:
It is kind of tiresome that we need two versions of this, what a pain. Maybe we should also have a dedicated slot for pre-dispatch proxy and get rid of the pre-dispatch-key mode stack.

Reply:
yep, I was kinda thinking the same (open question though if we want the dedicated PreDispatch slot to be a full faithful re-implementation of our Python key, or if it can still all live in python)